test/integration/sync.test.js

```diff
@@ -0,0 +1,363 @@
+runner.run(testFunc);
+
+function testFunc() {
+
+  const dataStoreTypes = [Kinvey.DataStoreType.Cache, Kinvey.DataStoreType.Sync];
+  let networkStore;
+  let syncStore;
+  let cacheStore;
+  let storeToTest;
+  const notFoundErrorName = 'NotFoundError';
+
+  //validates Push operation result for 1 created, 1 modified and 1 deleted locally items
+  const validatePushOperation = (result, createdItem, modifiedItem, deletedItem, expectedServerItemsCount) => {
+    return new Promise((resolve, reject) => {
+      expect(result.length).to.equal(3);
+      result.forEach((record) => {
+        expect(record.operation).to.equal(record._id === deletedItem._id ? 'DELETE' : 'PUT');
+        expect([createdItem._id, modifiedItem._id, deletedItem._id]).to.include(record._id);
+        if (record.operation !== 'DELETE') {
+          common.assertEntityMetadata(record.entity);
+          common.deleteEntityMetadata(record.entity);
+          expect(record.entity).to.deep.equal(record._id === createdItem._id ? createdItem : modifiedItem);
+        }
+        else {
+          expect(record.entity).to.not.exist;
+        }
+      })
+      networkStore.find().toPromise()
+        .then((result) => {
+          expect(result.length).to.equal(expectedServerItemsCount);
+          expect(_.find(result, (entity) => { return entity._id === deletedItem._id; })).to.not.exist;
+          expect(_.find(result, (entity) => { return entity.newProperty === modifiedItem.newProperty; })).to.exist;
+          let createdOnServer = _.find(result, (entity) => { return entity._id === createdItem._id; });
+
+          expect(common.deleteEntityMetadata(createdOnServer)).to.deep.equal(createdItem);
+          return storeToTest.pendingSyncCount()
+        })
+        .then((count) => {
+          expect(count).to.equal(0);
+          resolve();
+        }).catch(reject);
+    });
+  }
+  //validates Pull operation result
+  const validatePullOperation = (result, expectedItems, expectedPulledItemsCount) => {
+    return new Promise((resolve, reject) => {
```

Comment:
Same comments as the above function - redundant Promise construction. And the lambdas.